counter/models.py

```diff
@@ -309,6 +310,28 @@ def __str__(self):
         return "%s (%s)" % (self.name, self.code)
 
 
+class CounterQuerySet(models.QuerySet):
+    def annotate_is_barman(self, user: User) -> "CounterQuerySet":
+        """
+        Annotate the queryset with the `user_is_barman` field.
+        For each counter, this field has value True if the user
+        is a barman of this counter, else False.
+
+        :param user: the user we want to check if he is a barman
+
+        Example::
+
+            sli = User.objects.get(username="sli")
+            counters = Counter.objects.annotate_is_barman(sli).filter(user_is_barman=True)
+            print("Sli est barman dans les comptoirs suivants :")
+            for counter in counters:
+                print(f"- {counter.name}")
+        """
+        subquery = user.counters.filter(pk=OuterRef("pk"))
+        # noinspection PyTypeChecker
+        return self.annotate(user_is_barman=Exists(subquery))
```

Comment:
Je propose un meilleur nom pour ça
```suggestion
        return self.annotate(is_current_user_barman=Exists(subquery))
```

Reply:
On en a parlé, mais du coup non, ça décrit pas le comportement du queryset.